File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/parts/resource/MordekaiserShieldBarTest.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.IntSteps;
import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class MordekaiserShieldBarTest extends AbstractResourceTest {
    @Test
    void testMordekaiserShieldBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final MordekaiserShieldBar mordekaiserShieldBar = new MordekaiserShieldBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> mockResource(intSteps.nextInt(), 100))
                .testAnimation(mordekaiserShieldBar, 40);
    }
}

File: test/hud/parts/resource/MordekaiserShieldBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {MordekaiserShieldBar} from '../../../../src/hud/parts/resource/MordekaiserShieldBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Mordekaiser shield resource bar on actual hardware. */
describe('MordekaiserShield Integration', () => {
    setupResourceTest();

    test('testMordekaiserShieldBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const mordekaiserShieldBar = new MordekaiserShieldBar();

        await new AnimationTester()
            .withBeforeIterationAction(() => mockResource(intSteps.nextInt(), 100))
            .testAnimation(mordekaiserShieldBar, 40);
    }, 30000);
});
